BUILD.md

```diff
@@ -196,17 +199,29 @@ The following options control activating and locating varial options TPLs.
 * Kokkos_ENABLE_MEMKIND
     * Whether to enable the MEMKIND library
     * BOOL Default: Off
-* Kokkos_HPX_DIR
-    * Location of HPX library
+* Kokkos_ENABLE_LIBDL
+    * Whether to enable the LIBDL library
+    * BOOL Default: On
+* Kokkos_ENABLE_LIBRT
+    * Whether to enable the LIBRT library
+    * BOOL Default: Off
+* HPX_DIR or HPX_ROOT
```

Comment:
`Kokkos_HPX_DIR`?

Reply:
Yeah, this is confusing. I pushed a minor re-organization of the docs. Because HPX is a honest-to-goodness CMake package, `HPX_DIR` will configure which version of HPX gets found.

Anything prefixed by `Kokkos_` is telling *Kokkos* where to look for libraries. Anything without `Kokkos_` prefix is telling *CMake* where to find something with the standard `find_package` utility.

I didn't add all these details in `BUILD.md`. Hopefully the minor rework at least makes it clear why the HPX is different from the others.

I'm happy to restructure this, though, if you think different option names or more docs would make this clearer.